<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="base_name"         default="p1"/>
  <arg name="data_dir"          default="/mnt/md0/gear_data"/>
  <arg name="start_manager"     default="false"/>  
  <arg name="image_def"         default="hd"/>      
  <arg name="voxelize"          default="true"/>

  <arg name="nodelet_manager"   default="$(arg base_name)_pointcloud_logger_manager" if="$(arg start_manager)"/>
  <arg name="nodelet_manager"   default="$(arg base_name)" unless="$(arg start_manager)"/>     
  
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  
        args="manager" output="screen" if="$(arg start_manager)">
    <param name="num_worker_threads" value="4"/>
  </node>

  <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
  <node pkg="nodelet" type="nodelet" name="$(arg base_name)_voxel_grid" args="load pcl/VoxelGrid $(arg nodelet_manager)" 
    output="screen" if="$(arg voxelize)">
    <remap from="~input"                to="/$(arg base_name)/$(arg image_def)/points" />
    <remap from="~output"               to="/$(arg base_name)/$(arg image_def)/voxel_points" />
    <param name="leaf_size"             value="0.01"/>
    <param name="filter_limit_negative" value="true"/>
  </node>


  <!-- Point cloud logging node -->
  <node pkg="nodelet" type="nodelet" name="$(arg base_name)_points_logger"
        args="load gear_image_handler/PointcloudLogger $(arg nodelet_manager)" output="screen">
    <remap from="/log_pointcloud"     to="/$(arg base_name)/$(arg image_def)/voxel_points"  if="$(arg voxelize)"/>
    <remap from="/log_pointcloud"     to="/$(arg base_name)/$(arg image_def)/points"        unless="$(arg voxelize)"/>
    <remap from="/pointcloud_count"   to="/$(arg base_name)/pointcloud_count" />
    <param name="sensor_id"           value="$(arg base_name)"/>
    <param name="data_dir"            value="$(arg data_dir)"/>
    <param name="base_name"           value="$(arg base_name)"/>
  </node>

</launch>